zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -148,9 +149,12 @@ public boolean isValid(String id) {
   }
 
   /**
-   * Initialize a new ClientUriDomainMappingHelper instance if it hasn't been instantiated for the ACL provider.
+   * Initialize a new ClientUriDomainMappingHelper instance lazily if it hasn't been instantiated for the ACL provider.
```

Comment:
@desaikomal Yes that's true for all Lazy initialization with Double check locking design pattern.